app/bundles/CoreBundle/Event/EntityImportEvent.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+namespace Mautic\CoreBundle\Event;
+
+use Symfony\Contracts\EventDispatcher\Event;
+
+class EntityImportEvent extends Event
+{
+    /**
+     * @var array<int, int>
+     */
+    private array $idMap = [];
+
+    /**
+     * @var array<string, array<string, mixed>>
+     */
+    private array $dependencies = [];
+
+    public const UPDATE = 'update';
+    public const NEW    = 'new';
+    public const ERRORS = 'errors';
+
+    /**
+     * @var array<string, array<string, mixed>>
+     */
+    private array $status = [
+        self::UPDATE => [],
+        self::NEW    => [],
+        self::ERRORS => [],
+    ];
+
+    /** @phpstan-ignore-next-line */
+    public function __construct(private string $entityName, private array $data, private ?int $userId)
+    {
+    }
+
+    public function getEntityName(): string
+    {
+        return $this->entityName;
+    }
+
+    /** @phpstan-ignore-next-line */
+    public function getEntityData(): array
+    {
+        return $this->data;
+    }
+
+    public function getUserId(): ?int
+    {
+        return $this->userId;
+    }
+
+    /**
+     * Add a single entityIdMap.
+     */
+    public function addEntityIdMap(int $originalId, int $newId): void
+    {
+        $this->idMap[$originalId] = $newId;
+    }
+
+    /**
+     * Get IdMap.
+     *
+     * @return array<int, int>
+     */
+    public function getEntityIdMap(): array
+    {
+        return $this->idMap;
+    }
+
+    /**
+     * Get dependencies.
+     *
+     * @return array<string, array<string, mixed>>
+     */
+    public function getDependencies(): array
```

Comment:
Please remove comments that aren't adding any value. It's obvious what the method does even without the comment.
```suggestion
    /**
     * @return array<string, array<string, mixed>>
     */
    public function getDependencies(): array
```

Reply:
fixed